File: dagger-java/app/src/main/java/com/coder/dagger_java/MainActivity.java
package com.coder.dagger_java;

import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

import com.coder.dagger_java.data.DataManager;
import com.coder.dagger_java.data.model.User;
import com.coder.dagger_java.di.component.ActivityComponent;
import com.coder.dagger_java.di.component.DaggerActivityComponent;
import com.coder.dagger_java.di.module.ActivityModule;

import javax.inject.Inject;

public class MainActivity extends AppCompatActivity {

    @Inject
    DataManager mDataManager;

    private ActivityComponent activityComponent;

    private TextView mTvUserInfo;
    private TextView mTvAccessToken;

    public ActivityComponent getActivityComponent() {
        if (activityComponent == null) {
            activityComponent = DaggerActivityComponent.builder()
                    .activityModule(new ActivityModule(this))
                    .applicationComponent(DaggerApplication.get(this).getComponent())
                    .build();
        }
        return activityComponent;
    }


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        getActivityComponent().inject(this);

        mTvUserInfo = (TextView) findViewById(R.id.tv_user_info);
        mTvAccessToken = (TextView) findViewById(R.id.tv_access_token);
    }

    @Override
    protected void onPostCreate(@Nullable Bundle savedInstanceState) {
        super.onPostCreate(savedInstanceState);
        createUser();
        getUser();
        mDataManager.saveAccessToken("ASDR12443JFDJF43543J543H3K543");

        String token = mDataManager.getAccessToken();
        if (token != null) {
            mTvAccessToken.setText(token);
        }
    }

    private void createUser() {
        try {
            mDataManager.createUser(new User("Henna", "Castle Avenue, Dublin, Ireland"));
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    private void getUser() {
        try {
            User user = mDataManager.getUser(1L);
            mTvUserInfo.setText(user.toString());
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: dagger-kotlin/app/src/main/java/com/coder/dagger_kotlin/MainActivity.kt
package com.coder.dagger_kotlin

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import com.coder.dagger_kotlin.data.DataManager
import com.coder.dagger_kotlin.di.component.ActivityComponent
import com.coder.dagger_kotlin.di.component.DaggerActivityComponent
import com.coder.dagger_kotlin.di.module.ActivityModule
import javax.inject.Inject
import com.coder.dagger_kotlin.data.model.User
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {

    @Inject
    lateinit var dataManager: DataManager

    lateinit var activityComponent: ActivityComponent

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        activityComponent = DaggerActivityComponent.builder()
            .activityModule(ActivityModule(this))
            .applicationComponent(DaggerApplication.applicationComponent)
            .build()

        activityComponent.inject(this)
    }

    override fun onPostCreate(savedInstanceState: Bundle?) {
        super.onPostCreate(savedInstanceState)
        createUser()
        getUser()
        dataManager.saveAccessToken("ASDR12443JFDJF43543J543H3K543")

        val token = dataManager.getAccessToken()
        if (token != null) {
            tv_access_token.text = token
        }
    }

    private fun createUser() {
        try {
            dataManager.createUser(User(name = "Henna", address = "Castle Avenue, Dublin, Ireland"))
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    private fun getUser() {
        try {
            val user = dataManager.getUser(1)
            tv_user_info.text = user.toString()

        } catch (e: Exception) {
            e.printStackTrace()
        }

    }
}
